src/main/scala/lenthall/validation/ErrorOr.scala

```diff
@@ -19,4 +20,122 @@ object ErrorOr {
       }
     }
   }
+
+  // Note! See the bottom of this file for a generator function for all these near-identical ShortCircuitingFlatMapTupleNs...
+
+  implicit class ShortCircuitingFlatMapTuple1[A0](val t1: Tuple1[ErrorOr[A0]]) extends AnyVal {
+    def flatMapN[B](f_1: (A0) => ErrorOr[B]): ErrorOr[B] = t1._1.flatMap(f_1)
+  }
+
+  implicit class ShortCircuitingFlatMapTuple2[A0, A1](val t2: Tuple2[ErrorOr[A0], ErrorOr[A1]]) extends AnyVal {
+    def flatMapN[B](f_2: (A0, A1) => ErrorOr[B]): ErrorOr[B] = (t2._1, t2._2).tupled flatMap { case ((a, b)) => f_2(a, b) }
+  }
+
+  implicit class ShortCircuitingFlatMapTuple3[A0, A1, A2](val t3: Tuple3[ErrorOr[A0], ErrorOr[A1], ErrorOr[A2]]) extends AnyVal {
+    def flatMapN[B](f_3: (A0, A1, A2) => ErrorOr[B]): ErrorOr[B] = (t3._1, t3._2, t3._3).tupled flatMap { case ((a, b, c)) => f_3(a, b, c) }
+  }
```

Comment:
Not sure why you're unpacking the tuples only to put them back together.  Also the arg to `flatMap` can just be a `tupled` version of the function:

```
    def flatMapN[B](f_3: (A0, A1, A2) => ErrorOr[B]): ErrorOr[B] = t3.tupled flatMap f_3.tupled
```

Reply:
also why `t3` and `f_3` (with and without underscore)

Reply:
also starting to question the overall value of adding this code if it requires code generation and the logic can be expressed this succinctly

Reply:
...and if you call the parameters just `t` and `f` all the function bodies can be the same

Reply:
@mcovarr you're right, this can be succinctly replaced in code with
```
val x = (validation1, validation2).tupled flatMap { case (a, b) => validation3 }
```
All this provides is the slightly neater:
```
val x = (validation1, validation2) flatMapN { (a, b) => validation3 }
```
🤷‍♂️

Reply:
man_shrugging indeed. I really don't have super strong opinions about this, if you're writing code that's doing stuff like this and you find it useful then go for it.